Laba2SCCS/src/DataAccessLayer/Interfaces/IGenericRepository.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Interfaces
+{
+    public interface IGenericRepository<TModel>
+        where TModel : class
+    {
+        Task<IEnumerable<TModel>> GetAll();
```

Comment:
All methods that return Task should have a name with Async suffix (example GetAllAsync).
Rename it in ALL places (includes private methods in services if they return Task).